src/stream/mod.rs

```diff
@@ -880,7 +882,7 @@ pub trait Stream {
     /// This method is only available when the `use_std` feature of this
     /// library is activated, and it is activated by default.
     #[cfg(feature = "use_std")]
-    fn buffered(self, amt: usize) -> Buffered<Self>
+    fn buffered(self, amt: usize) -> Buffered<FuturesOrdered<<<Self as Stream>::Item as IntoFuture>::Future>, Self>
```

Comment:
I think the return types specifically could be papered over with a type alias.  The `FuturesSet` trait introduction is harder.  But does `FuturesUnordered` actually see a much direct use where consumers `::new()` one and then `push` to it manually?